hail/python/hail/linalg/blockmatrix.py

```diff
@@ -681,10 +681,14 @@ def write_from_entry_expr(entry_expr, path, overwrite=False, mean_impute=False,
         check_entry_indexed('BlockMatrix.write_from_entry_expr', entry_expr)
         mt = matrix_table_source('BlockMatrix.write_from_entry_expr', entry_expr)
 
-        if (not (mean_impute or center or normalize)) and (entry_expr in mt._fields_inverse):
-            #  FIXME: remove once select_entries on a field is free
-            field = mt._fields_inverse[entry_expr]
-            mt._write_block_matrix(path, overwrite, field, block_size)
+        if not (mean_impute or center or normalize):
+            if entry_expr in mt._fields_inverse:
+                #  FIXME: remove once select_entries on a field is free
+                field = mt._fields_inverse[entry_expr]
+                mt._write_block_matrix(path, overwrite, field, block_size)
```

Comment:
This prevents field pruning. Emitting a `mt.select_entries('field')` would fix that

Reply:
addressed. I also removed the FIXME in favor of a punch-list issue: https://github.com/hail-is/hail/issues/5202